### PR TITLE
Add axios api clients for apiv1 and the normal routes that return json

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: themes-default/slim/static/js/api.js [hunk 1]
```diff
@@ -1,6 +1,34 @@
...
+        'Content-Type': 'application/json',
+        'X-Api-Key': idToken
+    }
+});
+
+/**
+ * Api client based on the axios client, to communicate with medusa's api v1.
+ */
+const apiv1 = axios.create({ // eslint-disable-line no-unused-vars
+    baseURL: baseUrl.replace('api/v2/', 'api/v1/' + idToken + '/'),
```
Comment (OmgImAlexis) [inline themes-default/slim/static/js/api.js]: Any reason not to use the header like the other?

File: themes-default/slim/static/js/api.js [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+/**
+ * Api client based on the axios client, to communicate with medusa's web routes, which return json data.
+ */
+const apiRoute = axios.create({ // eslint-disable-line no-unused-vars
+    baseURL: baseUrl.replace('api/v2/', ''),
+    timeout: 10000,
+    headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        'X-Api-Key': idToken
```
[the remark this answers is not in the document]
Comment (p0psicles) [reply themes-default/slim/static/js/api.js]: Hmm I think @sharkykh assumes that the existing calls to the webroutes, also don't have the X-Api-Key in them. And that they just rely on the browsers basic authentication headers. I can try removing X-api-key.

[the remark this answers is not in the document]
Comment (OmgImAlexis) [reply themes-default/slim/static/js/api.js]: Oh true, forgot this was apiv1.

File: themes-default/slim/static/js/api.js (hunk 1 above)
Comment (p0psicles) [reply themes-default/slim/static/js/api.js]: I thought the api v1 does not use it?
And I checked apibuilder, and there it's adding the token to the query string?

File: themes-default/slim/static/js/api.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (p0psicles) [reply themes-default/slim/static/js/api.js]: No, this is not apiv1. This is just for basic webroutes. Like the `/home/getSeasonsSceneExceptions` call.

File: themes-default/slim/static/js/api.js (hunk 1 above)
Comment (OmgImAlexis) [reply themes-default/slim/static/js/api.js]: Ignore this as well. I got this and other other section mixed up.

File: themes-default/slim/static/js/api.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (p0psicles) [reply themes-default/slim/static/js/api.js]: You where right. x-api-key is not needed.

**Pull request merged**